std/conv.d

```diff
@@ -4075,8 +4077,11 @@ $(D T) is $(D @safe).
 Returns: A pointer to the newly constructed object.
  */
 T emplace(T, Args...)(void[] chunk, auto ref Args args)
-    if (is(T == class))
+if (is(T == class))
 {
+    static assert (!isAbstractClass!T, T.stringof ~
```

Comment:
All code inside contracts (even `static assert`) is ignored by the compiler in `release` mode. The other option is a template constrained, however they are used for overload resolution and the user would get an error like "none of the overloads of emplace is callable with..."